platform/core/src/services/UICustomizationService/UICustomizationService.ts

```diff
@@ -0,0 +1,273 @@
+  }
+
+  /** Mode customizations are changes to the behaviour of the extensions
+   * when running in a given mode.  Reset clears mode customizations.
+   * Note that global customizations over-ride mode customizations.
+   * @param defautlValue to return if no customization specified.
+   */
+  public getModeCustomization(
+    customizationId: string,
+    defaultValue?: UICustomization
+  ): UICustomization | void {
+    const customization =
+      this.globalCustomizations[customizationId] ??
+      this.modeCustomizations[customizationId] ??
+      defaultValue;
+    return this.applyUiType(customization);
+  }
+
+  /** Applies any inheritance due to UI Type customization */
+  public applyUiType(customization) {
+    if (!customization) return customization;
+    const { customizationType } = customization;
+    if (!customizationType) return customization;
+    const parent = this.getModeCustomization(customizationType);
+    return parent
+      ? Object.assign(Object.create(parent), customization)
+      : customization;
+  }
+
+  public addModeCustomizations(modeCustomizations): void {
+    if (!modeCustomizations) {
+      return;
+    }
+    this.addReferences(modeCustomizations, false);
+
+    this._broadcastModeCustomizationModified();
+  }
+
+  _broadcastModeCustomizationModified(): void {
+    this._broadcastEvent(EVENTS.MODE_CUSTOMIZATION_MODIFIED, {
+      modeCustomizations: this.modeCustomizations,
+      globalCustomizations: this.globalCustomizations,
+    });
+  }
+
+  /** Global customizations are those that affect parts of the GUI other than
+   * the modes.  They include things like settings for the search screen.
+   * Reset does NOT clear global customizations.
+   */
+  getGlobalCustomization(
+    id: string,
+    defaultValue?: UICustomization
+  ): UICustomization | void {
+    return this.applyUiType(this.globalCustomizations[id] ?? defaultValue);
+  }
+
+  setGlobalCustomization(id: string, value: UICustomization): void {
+    console.log('*** Set global', id, value);
+    this.globalCustomizations[id] = value;
+    this._broadcastGlobalCustomizationModified();
+  }
+
+  protected setConfigGlobalCustomization(configuration: UIConfiguration): void {
+    this.globalCustomizations = {};
+    const keys = flattenNestedStrings(configuration.globalCustomizations);
+    this.readCustomizationTypes(
+      v => keys[v.name] && v.customization,
+      this.globalCustomizations
+    );
+
+    // TODO - iterate over customizations, loading them from the extension
+    // manager.
+    this._broadcastGlobalCustomizationModified();
+  }
+
+  _broadcastGlobalCustomizationModified(): void {
+    this._broadcastEvent(EVENTS.GLOBAL_CUSTOMIZATION_MODIFIED, {
+      modeCustomizations: this.modeCustomizations,
+      globalCustomizations: this.globalCustomizations,
+    });
+  }
+
+  findExtensionValue(value: string) {
+    const entry = this.extensionManager.getModuleEntry(value);
+    return entry;
+  }
+
+  initDefaults() {
```

Comment:
I see the order of methods in this class is not ideal. For instance, this function is called in the `init` but it is located almost at the bottom of the class. Function calls that are related should be almost close together with the caller being on top of callee.

Reply:
Updated, but see if you think more changes are necessary.